src/librustc_codegen_llvm/base.rs

```diff
@@ -168,10 +170,58 @@ pub fn compile_codegen_unit(
                 cx.create_used_variable()
             }
 
+            if cx.has_debug() {
+                if let Some(sir_cx) = cx.sir_cx.borrow_mut().as_mut() {
+                    for (func_idx, blocks) in sir_cx.funcs_and_blocks_deterministic()
+                        .iter_enumerated()
+                    {
+                        let sir_func = &sir_cx.funcs[func_idx];
+                        let sym_name = &sir_func.symbol_name;
+
+                        // Don't label the main wrapper as it causes label insertion to crash and
+                        // is never a candidate for tracing.
+                        if sym_name == "main" {
+                            continue;
+                        }
+
+                        // When --test is passed, the auto-generated entry point causes our label
+                        // insertion to crash. We never want to trace test harnesses anyway, so
+                        // skip them.
+                        if tcx.sess.opts.test {
+                            let entry_fn = tcx.entry_fn(LOCAL_CRATE).unwrap().0;
+                            let entry_inst = Instance::mono(tcx, entry_fn);
+                            let entry_sym = &*tcx.symbol_name(entry_inst).name.as_str();
+                            if sym_name == entry_sym {
+                                dbg!("skipping a test harness main");
+                                continue;
+                            }
+                        }
+
+                        for (bb_idx, bb) in blocks.iter_enumerated() {
+                            let llbb = unsafe {
+                                &*(*bb as *const sir::BasicBlock as *const llvm::BasicBlock)
+                            };
+
+                            let lbl_name = CString::new(format!("__YK_BLK:{}:{}",
```

Comment:
Should the `__YK_BLK` be a constant somewhere? I guess we use this in more than one place?

Reply:
Currently we only use this here since moving label generation closer to LLVM. And I don't think we'll be adding blocks anywhere else. That being said, I'm not generally opposed to putting this in a constant.

Reply:
@ptersilie

Reply:
Empty comment from @vext01?

Reply:
I was prompting @ptersilie to answer, because I wasn't sure :P

Reply:
@ptersilie the mapper in `yk` will need this constant too, so perhaps it should be a constant in yk? I can do that if you like.